src/plugins/tracker.ts

```diff
@@ -0,0 +1,85 @@
+import ABTestingService from "@/plugins/ab-testing-service.js"
+
+declare global {
+  interface Window {
+    _paq?: [
+      string | (() => void),
+      (number | string)?,
+      string?,
+      string?,
+      string?
+    ][]
+  }
+}
+
+// * Waits for `window._paq` to become available.
+//   Stops trying after reaching the specified timeout.
+function waitForPaq(timeout = 10000): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const startTime = Date.now()
+
+    const intervalId = setInterval(() => {
+      if (window._paq) {
+        clearInterval(intervalId)
+        resolve()
+      } else if (Date.now() - startTime > timeout) {
+        clearInterval(intervalId)
+        reject(new Error("Timeout reached."))
+      }
+    }, 100)
+  })
+}
+
+function initializeABTestingDimensions() {
+  const ABTestingEnvironment = ABTestingService.getEnvironment()
+  for (const name in ABTestingEnvironment) {
+    const data = ABTestingEnvironment[name]
+    if (data.deleted) {
+      tracker.deleteCustomDimension(data.index)
+    } else {
+      tracker.setCustomDimension(data.index, `${name}/${data.value}`)
+    }
+  }
+}
+
+waitForPaq().then(() => {
+  initializeABTestingDimensions()
+})
+
+const tracker = {
+  trackEvent: (
+    category: string,
+    action: string,
+    name?: string,
+    value?: string
+  ) => {
+    window._paq?.push(["trackEvent", category, action, name, value])
+  },
+  disableTracking: async () => {
+    window._paq?.push(["optUserOut"])
+  },
+  enableTracking: async () => {
+    window._paq?.push(["forgetUserOptOut"])
+  },
+  pushCallback: async (callback: () => void) => {
+    window._paq?.push([callback])
+  },
+  deleteCustomDimension: async (index: number) => {
+    window._paq?.push(["deleteCustomDimension", index])
+  },
+  setCustomDimension: async (index: number, value: string) => {
+    window._paq?.push(["setCustomDimension", index, value])
+  },
+  getVisitorId: async () => {
+    let visitorId: string | undefined
+    window._paq?.push([
+      function () {
+        visitorId = this?.getVisitorId()
+      },
+    ])
+
+    return visitorId
```

Comment:
Pas bloquant non plus :
```suggestion
  return new Promise<string | undefined>((resolve) => {
    window._paq?.push([() => resolve(window._paq?.getVisitorId?.())])
  })
```

Reply:
Du coup j'ai vérifié on a bien tout en synchrone, j'ai viré le `async`